From 3e98836b52b12f08267889d5538d7c6c85d5df40 Mon Sep 17 00:00:00 2001
From: Nicholas Lim <nicknitewolf@lineageos.org>
Date: Tue, 24 Sep 2019 19:59:46 +0800
Subject: [PATCH] Revert "ART: Add version check for memfd_create()"

 * This check is redundant as devices that lack
   the memfd_creat syscall will fail to boot
   regardless of the check.

This reverts commit 51f89d9b465e34fb18a5704c6dfe885ebad817d0.

Change-Id: Ife2cebde713e14bd787f48628092650866e2ac09
Signed-off-by: Satish Patel <tosatishpatel@gmail.com>
---
 libartbase/base/memfd.cc | 16 ----------------
 libartbase/base/memfd.h  |  3 +--
 2 files changed, 1 insertion(+), 18 deletions(-)

diff --git a/libartbase/base/memfd.cc b/libartbase/base/memfd.cc
index 780be328af..e57c948e8d 100644
--- a/libartbase/base/memfd.cc
+++ b/libartbase/base/memfd.cc
@@ -17,10 +17,8 @@
 #include "memfd.h"
 
 #include <errno.h>
-#include <stdio.h>
 #if !defined(_WIN32)
 #include <sys/syscall.h>
-#include <sys/utsname.h>
 #include <unistd.h>
 #endif
 
@@ -41,20 +39,6 @@ namespace art {
 #if defined(__NR_memfd_create)
 
 int memfd_create(const char* name, unsigned int flags) {
-  // Check kernel version supports memfd_create(). Some older kernels segfault executing
-  // memfd_create() rather than returning ENOSYS (b/116769556).
-  static constexpr int kRequiredMajor = 3;
-  static constexpr int kRequiredMinor = 17;
-  struct utsname uts;
-  int major, minor;
-  if (uname(&uts) != 0 ||
-      strcmp(uts.sysname, "Linux") != 0 ||
-      sscanf(uts.release, "%d.%d", &major, &minor) != 2 ||
-      (major < kRequiredMajor || (major == kRequiredMajor && minor < kRequiredMinor))) {
-    errno = ENOSYS;
-    return -1;
-  }
-
   return syscall(__NR_memfd_create, name, flags);
 }
 
diff --git a/libartbase/base/memfd.h b/libartbase/base/memfd.h
index 91db0b2d8f..4367198185 100644
--- a/libartbase/base/memfd.h
+++ b/libartbase/base/memfd.h
@@ -19,8 +19,7 @@
 
 namespace art {
 
-// Call memfd(2) if available on platform and return result. This call also makes a kernel version
-// check for safety on older kernels (b/116769556)..
+  // Call memfd(2) if available on platform and return result.
 int memfd_create(const char* name, unsigned int flags);
 
 }  // namespace art
-- 
2.17.1

